compiler/luci/pass/src/FuseGRUPass.cpp

```diff
@@ -0,0 +1,674 @@
+ *            |
+ *            V
+ *          [Out]
+ *
+ * Condition Graph:
+ *          [In]   [CircleConst] (scalar int32 value)
+ *           |        |
+ *           V        |
+ *         [Less]------
+ *           |
+ *           V
+ *         [Out]
+ *
+ *  Body Graph must contain:
+ *  - 2 CircleFullyConnected nodes;
+ *  - 3 CircleMul nodes;
+ *  - 2 CircleLogistic nodes;
+ *  - 2 CircleSplit nodes;
+ *  - 6 CircleAdd nodes;
+ *  - 1 CircleGather node;
+ *  - 1 CircleReshape node;
+ *  - 1 CircleSub node;
+ *  - 1 CircleTanh node;
+ *  - 6 CircleSplitOut nodes;
+ *  - 5 CircleInput nodes;
+ *  - 5 CircleOutput nodes;
+ *
+ *  Body Graph:
+ *  [In_1]                     [In_2]--->[Add_2 (with Const)]--->[Out_2]            [In_3]
+ *    |    \                                                      |                   |
+ *    |     \                                         [In_4]---[Gather]      [Add_3 (with Const)]
+ *    |     [FullyConnected_1]                          |         |                   |
+ *    |               |                              [Out_4]      |                [Out_3]
+ *    |           [Split_1]                                 [FullyConnected_2]
+ *    |         /         |   \                                       |
+ *    |        |          |    \                                   [Split_2]
+ *    |    [Add_1] -------+----+---------------------------------/    |     |
+ *    |         |         |    |                                      |     |
+ *    |         |         |    ------------------------------------[Add_4]  |
+ *    |         |         |                                           |     |
+ *    |         |         |                                   [Logistic_1]  |
+ *    |         |         |                                          |      |
+ *    |         |         ----------------------------------------[Mul_2]   |
+ *    |         |                                                    \      /
+ *    |         |                                                     [Add_5]
+ *    |         |                                                        |
+ *    |      [Logistic_2]                                              [Tanh]
+ *     \     /           \                                               |
+ *      [Mul_1]          [Sub (with const)]                              |
+ *       \                                 \                             |
+ *        \                                 ---------------------------[Mul_3]
+ *         \                                                           /
+ *          \                                                         /
+ *           --------------------[Add_6]------------------------------
+ *                              /     \
+ *                             /       \
+ *                         [Reshape]   [Out_5]
+ *                             |
+ *                           [Out_1]
+ */
+class GRUPattern1 final : public GRUPatternBase
```

Comment:
I assume `1` suffix in `GRUPattern1` is to prepare more patterns.
Please leave a note about thus.
Or please use just `GRUPattern` here and later we can rename this when we add more patterns.